scripts/viewer/run_viewer.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python
+"""
+Starts viewer in eval mode.
+"""
+from __future__ import annotations
+
+import time
+from dataclasses import dataclass, fields
+from pathlib import Path
+
+import tyro
+from rich.console import Console
+
+from nerfstudio.configs.base_config import ViewerConfig
+from nerfstudio.engine.trainer import TrainerConfig
+from nerfstudio.pipelines.base_pipeline import Pipeline
+from nerfstudio.utils.eval_utils import eval_setup
+from nerfstudio.viewer.server import viewer_utils
+
+CONSOLE = Console(width=120)
+
+
+@dataclass
+class ViewerConfigWithoutNumRays(ViewerConfig):
+    """Configuration for viewer instantiation"""
+
+    num_rays_per_chunk: tyro.conf.Suppress[int] = -1
+    start_train: tyro.conf.Suppress[bool] = False
+
+    def as_viewer_config(self):
+        """Converts the instance to ViewerConfig"""
+        return ViewerConfig(**{x.name: getattr(self, x.name) for x in fields(self)})
+
+
+@dataclass(frozen=True)
+class RunViewer:
+    """Load a checkpoint and start the viewer."""
+
+    load_config: Path
+    """Path to config YAML file."""
+    viewer: ViewerConfigWithoutNumRays = ViewerConfigWithoutNumRays()
+    """Viewer configuration"""
+
+    def main(self) -> None:
+        """Main function."""
+        config, pipeline, _ = eval_setup(
+            self.load_config,
+            eval_num_rays_per_chunk=None,
+            test_mode="test",
+        )
+        num_rays_per_chunk = config.viewer.num_rays_per_chunk
+        assert self.viewer.num_rays_per_chunk == -1
+        config.vis = "viewer"
+        config.viewer = self.viewer.as_viewer_config()
+        config.viewer.num_rays_per_chunk = num_rays_per_chunk
+
+        self._start_viewer(config, pipeline)
+
+    def _start_viewer(self, config: TrainerConfig, pipeline):
+        base_dir = config.get_base_dir()
+        viewer_log_path = base_dir / config.viewer.relative_log_filename
+        viewer_state = None
+        viewer_state, _ = viewer_utils.setup_viewer(
```

Comment:
I have removed the `writer.setup_local_writer`, but I had to mock `writer.GLOBAL_BUFFER`. Is it ok?

Reply:
Hmm maybe it is necessary, we use it to store `EventName.VIS_RAYS_PER_SEC` which is used to figure out the resolution of the render.

Reply:
We can still store these events without setup_local_writer by manually initialising the GLOBAL_BUFFER. What is the reason for avoiding setup_local_writer?

Reply:
No reason, I just thought it was only necessary for training and therefore not needed here. But since it is also used for other things, we can keep it around.

Reply:
I added it back.